enterprise/replicasync/replicasync.go

```diff
@@ -466,3 +477,28 @@ func (m *Manager) Close() error {
 	}
 	return nil
 }
+
+func CreateDERPMeshTLSConfig(hostname string, tlsCertificates []tls.Certificate) (*tls.Config, error) {
+	meshRootCA := x509.NewCertPool()
+	for _, certificate := range tlsCertificates {
+		for _, certificatePart := range certificate.Certificate {
+			certificate, err := x509.ParseCertificate(certificatePart)
+			if err != nil {
+				return nil, xerrors.Errorf("parse certificate %s: %w", certificate.Subject.CommonName, err)
+			}
+			meshRootCA.AddCert(certificate)
+		}
+	}
+	// This TLS configuration spoofs access from the access URL hostname
+	// assuming that the certificates provided will cover that hostname.
+	//
+	// Replica sync and DERP meshing require accessing replicas via their
+	// internal IP addresses, and if TLS is configured we use the same
+	// certificates.
+	return &tls.Config{
+		MinVersion:   tls.VersionTLS12,
+		Certificates: tlsCertificates,
```

Comment:
This whole block of code is copied from coderd to avoid duplicating it. We use the TLS server certs as the expected certs on the TLS client when doing DERP mesh requests because that's the cert we expect to see on other replicas too.

Reply:
`Certificates` are the certificates that we send.  My point is that we don't use this TLS config to send any certificates.

  The certificates that we expect to see on the other replicas are configured in the `RootCAs`.

Reply:
You're right, we can remove Certificates. I'm not sure why this was here originally in that case lol - I'll also rewrite all the comments